src/workerd/util/sqlite.c++

```diff
@@ -61,6 +62,41 @@ namespace workerd {
 // Version of `SQLITE_CALL` that can be called after inspecting the error code, in case some codes
 // aren't really errors.
 
+#define VALUE_GATE(variable, maybeNull, ...) do { \
+    bool valid = false;                           \
+    if (maybeNull == nullptr) { \
+      KJ_IF_MAYBE(value, variable) { \
+        const std::string validValues[] = { __VA_ARGS__ }; \
```

Comment:
Using `std::string` here means each of the inputs will cause a heap allocation in order to copy the string to the heap. You should use `kj::StringPtr` instead, which simply points at the existing value.

In general, we do not use `std::string` in KJ code, because it has quite a few footguns, especially around implicit memory allocation when you don't want it, as well as historically some atomic reference counting "optimizations" that turned out to actually be pessimizations (though I think those were later removed).